contracts/SignerBondsUniswapV2.sol

```diff
@@ -75,13 +76,22 @@ contract SignerBondsUniswapV2 is ISignerBondsSwapStrategy, Ownable {
     // Determines the deadline in which the swap transaction has to be mined.
     // If that deadline is exceeded, transaction will be reverted.
     uint256 public swapDeadline = 20 minutes;
+    // Determines if the open auctions check should be performed. If true,
+    // swaps cannot be performed if open auctions count is bigger than zero.
+    // If false, open auctions are not taken into account.
+    bool public openAuctionsCheck = true;
 
     event UniswapV2SwapExecuted(uint256[] amounts);
 
-    constructor(IUniswapV2Router _uniswapRouter, CoveragePool _coveragePool) {
+    constructor(
+        IUniswapV2Router _uniswapRouter,
+        CoveragePool _coveragePool,
+        Auctioneer _auctioneer
```

Comment:
I think this cyclic dependency Tomasz mentioned is a symptom of a bigger problem with how we do swaps. Specifically, `swapSignerBonds` is not executing a swap but just depositing funds on a particular strategy. For this reason, there is no point to require no auction is open before this method is called. Secondly, what happens if most of the liquidity is removed from Uniswap v2 and governance was not able to swap all assets? ETH will be locked forever in Uniswap V2 strategy...

What I think we could do:
- Let's not call `swapSignerBonds` from the risk manager. Let's accumulate all ETH on the risk manager.
- Let's add a `withdrawBonds(uint256 amount)` function to the risk manager that can be called only by the current strategy.
- Let's not pass `_signerBondsSwapStrategy` as a constructor parameter of `CoveragePool`. We do not need the strategy to be set right away for the risk manager to work in this model, it can be set later.
- (Optional) instead of the constructor, we can `initialize` Risk Manager so that we do not have the cyclic dependency and we can approve right away.

Reply:
Please see the new approach: https://github.com/keep-network/coverage-pools/pull/88/commits/3f36835113dac64bdd05101fd10fd69b0c143f3a